crates/youki/src/main.rs

```diff
@@ -145,15 +145,18 @@ fn determine_root_path(root_path: Option<PathBuf>) -> Result<PathBuf> {
     }
 
     // see https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    let uid = getuid().as_raw();
     if let Ok(path) = std::env::var("XDG_RUNTIME_DIR") {
-        return Ok(PathBuf::from(path));
+        let path = Path::new(&path).join("youki");
+        if create_dir_all_with_mode(&path, uid, Mode::S_IRWXU).is_ok() {
+            return Ok(path);
+        }
```

Comment:
If it fails, it must return with an error or the process will continue, which is what you are aiming for, right?

Reply:
Well, not immediately.  If this create fails we continue onto the remaining options (`$HOME/.youki/run` then `/tmp/youki/<uid>`).  AFAICT that's what happened before, and I'm not intending to change that.

Reply:
That is correct. If we cannot get the preferred location (which is XDG_RUNTIME_DIR), we try a bunch of other locations that may be suitable before we give up.